### ISSUE TITLE
Is replacing an enum with a typedef semver-major or just a hazard?

### ISSUE THREAD

**Issue by obi1kenobi** [obi1kenobi/cargo-semver-checks#413]
Replacing an enum with a typedef is a breaking change, due to the inability to glob-import via the typedef.

But [not all breaking changes are major](https://predr.ag/blog/some-rust-breaking-changes-do-not-require-major-version/) — so is this one major? Is it minor? Should it instead just be a `warn` level lint pointing out the hazard?

[Playground link](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=0195f06da29fa4659dfe9e9e5a77c4c0)
```rust
fn main() {
    // Works
    // enum Foo {A, B}

    enum SecretFoo {
        A,
        B,
    }
    type Foo = SecretFoo;
    
    // Works fine
    let x = Foo::A;

    // Doesn't work
    use Foo::*;
    let y = A;
}
```

Example courtesy of @aDotInTheVoid.

**Comment by obi1kenobi** [comment]
It's not just glob imports that are problematic. Replacing the glob with a direct import of the variant (`use Foo::A`) doesn't work either.

**Comment by obi1kenobi** [comment]
I believe this is just a hazard, not breaking. I've laid out my reasoning here: https://predr.ag/blog/re-exporting-enum-with-type-alias-breaking-change-not-major/

**Comment by obi1kenobi** [comment]
This issue just happened in the `time` crate in v0.3.35: https://github.com/time-rs/time/issues/675
